### DISCUSSION TITLE
Cosim throwing segmentation fault

### DISCUSSION THREAD

**Discussion by mysoreanoop** [chipsalliance/dromajo#61] [Q&A] [answered]
Hi,
I'm trying to set up cosim with in a project, I have the attached cosim host code, and a dummy DUT input. 
The dromajo_cosim_init() call is throwing a segmentation fault, and it's unclear what the reason is.

The code was run with:
g++ ps.cpp -L<lib_dir> -ldromajo_cosim -I<dromajo_cosim.h dir> -lpthread -o ps.out
[ps.cpp.txt](https://github.com/chipsalliance/dromajo/files/8423806/ps.cpp.txt)
[qsort.riscv.txt](https://github.com/chipsalliance/dromajo/files/8423812/qsort.riscv.txt)
t

The error I see is:
Initing                                        --> my print
Segmentation fault

**Comment by et-tommythorn** [comment, marked as answer]
Well, that's not very much to go on.  The usual way to debug things would be a start: build with symbols (-g) and run under the debugger.

**Reply by mysoreanoop** [reply]
Yes, sorry, I did that and later found out the issue was that I was passing in wrong argc for the argv passed. Thanks nonetheless.
